### PR TITLE
Implemented noit_metric_rollup

### PR DESCRIPTION
**Pull request by JonasKunze**
renamed metric_id_t -> noit_metric_id_t
Pushing noit_metric_message_t.value as lightuserdata so that the data mapping can be done via ffi_libnoit.lua

### CODE REVIEW

File: src/Makefile.in [hunk 1]
```diff
@@ -261,6 +261,10 @@ stratcon_datastore.o:	stratcon_datastore.c
 		echo "- compiling $<" ; \
 		$(CC) $(CPPFLAGS) $(CFLAGS) -c $< ; \
 	fi
+	
+noit_metric_rollup.o:  
+  $(CC) -fPIC $(CPPFLAGS) $(CFLAGS) -c noit_metric_rollup.c
+
```
Comment (postwait) [inline src/Makefile.in]: Why is this needed?  We have a generic object building rule

Comment (JonasKunze) [inline src/Makefile.in]: I needed -fPIC but I accidentally built the .o files instead of the .lo one
Now it's not needed any more

File: src/Makefile.in [hunk 2]
```diff
@@ -261,7 +261,7 @@ stratcon_datastore.o:	stratcon_datastore.c
 		echo "- compiling $<" ; \
 		$(CC) $(CPPFLAGS) $(CFLAGS) -c $< ; \
 	fi
-
+
```
Comment (postwait) [inline src/Makefile.in]: loose tab here

File: src/modules/noit_lua_noit_binding.c [hunk 3]
```diff
@@ -138,89 +139,10 @@ lua_noit_checks_unsubscribe(lua_State *L) {
...
-            lua_pushinteger(L, metric_id->value.v_int32);
-            break;
-          case METRIC_INT64:
-            lua_pushinteger(L, metric_id->value.v_int64);
-            break;
-          case METRIC_UINT32:
-            lua_pushinteger(L, metric_id->value.v_int32);
-            break;
-          case METRIC_UINT64:
-            lua_pushinteger(L, metric_id->value.v_int64);
```
Comment (postwait) [inline src/modules/noit_lua_noit_binding.c]: This should check to see if it represents a negative number in two's complement (would role) and if so hand it as a double into lua.

File: src/modules/noit_lua_noit_binding.c [hunk 4]
```diff
@@ -303,9 +225,16 @@ noit_message_index_func(lua_State *L) {
         break;
       }
       return 1;
+    case 'w':
+      if(!strcmp(k, "whence")) {
+        lua_pushinteger(L, msg->value.whence_ms/1000);
```
Comment (postwait) [inline src/modules/noit_lua_noit_binding.c]: is there a reason we hand this back as integer seconds?  That seem incongruent with the way reconnoiter looks at times (which are either timevals or milliseconds since epoch).  I think this should come back as either milliseconds or as two return values: seconds and microseconds (like timevals)

File: src/modules/noit_lua_noit_binding.c (hunk 3 above)
Comment (JonasKunze) [inline src/modules/noit_lua_noit_binding.c]: that code is deleted! I've replaced it with ffi_libnoit.lua:

I'm using it this way:
`
self.numeric = ffi.new("noit_numeric_rollup_accu", {})

...

if self.numeric.accumulated.type == string.byte("i") then
    value =  self.numeric.accumulated.value.v_int32
  end

  if self.numeric.accumulated.type == string.byte("l") then
    value = self.numeric.accumulated.value.v_int64
  end

  if self.numeric.accumulated.type == string.byte("I") then
    value = self.numeric.accumulated.value.v_int32
  end

  if self.numeric.accumulated.type == string.byte("L") then
    value = self.numeric.accumulated.value.v_int32
  end

  if self.numeric.accumulated.type == string.byte("n") then
    value = self.numeric.accumulated.value.v_double
  end

  if self.numeric.accumulated.type == string.byte("s") then
    value = self.numeric.accumulated.value.v_string
  end
  if value ~= false then
    return tonumber(value)
  end
  return false
`

Comment (postwait) [inline src/modules/noit_lua_noit_binding.c]: sorry misread... Also there's a bug in your example for "L"... just be careful :-)

File: src/modules/noit_lua_noit_binding.c (hunk 4 above)
Comment (JonasKunze) [inline src/modules/noit_lua_noit_binding.c]: I had the fear that working with large numbes (milliseconds) in lua could lead to rounding errors. But in the end 53 bit are more than enough for unix time in milliseconds. I'll change this to `whence_ms` again

File: src/modules/noit_lua_noit_binding.c (hunk 3 above)
Comment (JonasKunze) [inline src/modules/noit_lua_noit_binding.c]: thx!

**Comment by postwait** [comment]
LGTM
